listener/Dockerfile

```diff
@@ -0,0 +1,199 @@
+    python3-pip \
+    python3-toml \
+    python3-requests \
+    python3-colorlog \
+    python3-twisted \
+    swig \
+    software-properties-common \
+    curl \
+    dh-autoreconf \
+    ocaml \
+    wget \
+    xxd \
+    unzip \
+    tar \
+    ocamlbuild \
+    python3-tk \
+    python3-pycodestyle \
+    pycodestyle \
+    pylint3 \
+    cppcheck \
+ && apt-get -y -q upgrade \
+ && apt-get clean \
+ && rm -rf /var/lib/apt/lists/*
+
+# Install setuptools packages using pip because
+# these are not available in apt repository.
+RUN pip3 install --upgrade setuptools json-rpc nose2
+
+# Make Python3 default
+RUN ln -sf /usr/bin/python3 /usr/bin/python
+
+# SGX common library and SDK are installed in /opt/intel directory.
+# Installation of SGX libsgx-common packages requires /etc/init directory. In docker image this directory doesn't exist.
+# Hence creating /etc/init directory.
+RUN mkdir -p /opt/intel \
+ && mkdir -p /etc/init
+WORKDIR /opt/intel
+
+# Install SGX common library
+RUN wget https://download.01.org/intel-sgx/linux-2.3.1/ubuntu18.04/libsgx-enclave-common_2.3.101.46683-1_amd64.deb \
+ && dpkg -i libsgx-enclave-common_2.3.101.46683-1_amd64.deb
+
+# Install SGX SDK
+RUN wget https://download.01.org/intel-sgx/linux-2.3.1/ubuntu18.04/sgx_linux_x64_sdk_2.3.101.46683.bin \
+ && chmod +x sgx_linux_x64_sdk_2.3.101.46683.bin \
+ && echo "yes" | ./sgx_linux_x64_sdk_2.3.101.46683.bin \
+ && rm sgx_linux_x64_sdk_2.3.101.46683.bin \
+ && echo ". /opt/intel/sgxsdk/environment" >> /etc/environment
+
+WORKDIR /tmp
+
+# Build ("Untrusted") OpenSSL, then build ("trusted") SGX OpenSSL
+# Note: The latter will compile in HW or SIM mode depending on the
+# availability of /dev/isgx and /var/run/aesmd/aesm.socket
+RUN OPENSSL_VER=1.1.1d \
+ && wget https://www.openssl.org/source/openssl-$OPENSSL_VER.tar.gz \
+ && tar -zxf openssl-$OPENSSL_VER.tar.gz \
+ && cd openssl-$OPENSSL_VER/ \
+ && ./config \
+ && THREADS=8 \
+ && make -j$THREADS \
+ && make test \
+ && make install -j$THREADS \
+ && ldconfig \
+ && ln -s /etc/ssl/certs/* /usr/local/ssl/certs/ \
+ && cd .. \
+ && rm -rf openssl-$OPENSSL_VER \
+ && git clone https://github.com/intel/intel-sgx-ssl.git  \
+ && . /opt/intel/sgxsdk/environment \
+ && (cd intel-sgx-ssl/openssl_source; mv /tmp/openssl-$OPENSSL_VER.tar.gz . ) \
+ && (cd intel-sgx-ssl/Linux; \
+    if ([ -c /dev/isgx ] && [ -S /var/run/aesmd/aesm.socket ]); then SGX_MODE=HW; \
+    else SGX_MODE=SIM; \
+    fi; \
+    make SGX_MODE=${SGX_MODE} DESTDIR=/opt/intel/sgxssl all test ) \
+ && (cd intel-sgx-ssl/Linux; make install ) \
+ && rm -rf /tmp/intel-sgx-ssl \
+ && echo "SGX_SSL=/opt/intel/sgxssl" >> /etc/environment
+
+COPY . /project/avalon
+
+ARG SGX_MODE
+ARG MAKECLEAN
+ARG TCF_DEBUG_BUILD
+ARG DISPLAY
+ARG XAUTHORITY
+
+# Environment setup
+
+RUN echo "TCF_HOME=/project/avalon/" >> /etc/environment \
+ && if [ ! -z "$http_proxy"  ]; then \
+    echo 'Acquire::http::Proxy "'$http_proxy'";' >> /etc/apt/apt.conf.d/00proxy; \

```

Comment:
Do we need set http_proxy explicitly? we are passing as argument to docker-compose that will be taken care right?

Reply:
ok